pkg/core/native/native_test/neo_test.go

```diff
@@ -271,6 +271,33 @@ func TestNEO_Vote(t *testing.T) {
 		require.NotEqual(t, candidates[0], pubs[i])
 		require.NotEqual(t, candidates[len(candidates)-1], pubs[i])
 	}
+	t.Run("LastGasPerVote", func(t *testing.T) {
+		getAccountState := func(t *testing.T, account util.Uint160) *state.NEOBalance {
+			stack, err := neoCommitteeInvoker.TestInvoke(t, "getAccountState", account)
+			require.NoError(t, err)
+			as := new(state.NEOBalance)
+			err = as.FromStackItem(stack.Pop().Item())
+			require.NoError(t, err)
+			return as
+		}
+		registerTx = neoValidatorsInvoker.WithSigners(candidates[0]).PrepareInvoke(t, "registerCandidate", candidates[0].(neotest.SingleSigner).Account().PublicKey().Bytes())
+		voteTx = neoValidatorsInvoker.WithSigners(voters[0]).PrepareInvoke(t, "vote", voters[0].(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), candidates[0].(neotest.SingleSigner).Account().PublicKey().Bytes())
+		neoValidatorsInvoker.AddNewBlock(t, registerTx, voteTx)
+		e.CheckHalt(t, registerTx.Hash(), stackitem.Make(true))
+		e.CheckHalt(t, voteTx.Hash(), stackitem.Make(true))
+
+		stateBeforeUnvote := getAccountState(t, voters[0].ScriptHash())
+		require.NotEqual(t, uint64(0), stateBeforeUnvote.LastGasPerVote.Uint64())
+		// unvote
+		unvoteTx := neoValidatorsInvoker.WithSigners(voters[0]).PrepareInvoke(t, "vote", voters[0].(neotest.SingleSigner).Account().PrivateKey().GetScriptHash(), nil)
+		neoValidatorsInvoker.AddNewBlock(t, unvoteTx)
+		e.CheckHalt(t, unvoteTx.Hash(), stackitem.Make(true))
+		advanceChain(t)
+
+		stateAfterUnvote := getAccountState(t, voters[0].ScriptHash())
```

Comment:
`getAccountState` is not enough, you also need to ensure that resulting stackitem contains 4 subitems, and the fourth one is zero integer, because `(s *NEOBalance) FromStackItem` is able to handle both 3 and 4 subitems.